presentation-api/controlling-ua/PresentationRequest_onconnectionavailable-manual.html

```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>PresentationRequest.onconnectionavailable (manual test)</title>
+<link rel="author" title="Tomoyuki Shimizu" href="https://github.com/tomoyukilabs">
+<link rel="help" href="https://w3c.github.io/presentation-api/#starting-a-presentation">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<p>Click the button below and select the available presentation display, to start the manual test.</p>
+<button id="presentBtn" onclick="startPresentation()">Start Presentation Test</button>
+
+
+<script>
+    // ----------
+    // DOM Object
+    // ----------
+    var presentBtn = document.getElementById("presentBtn");
+
+    // ------------
+    // Request init
+    // ------------
+    var validUnixDate = new Date().getTime() + String(Math.floor(Math.random() * 1e5)),
+            validPresURL = '../receiving-ua/idlharness.html#__castAppId__=2334D33A/__castClientId__=' + validUnixDate,
+            request = new PresentationRequest(validPresURL);
+
+    // --------------------------------------------------------------------------
+    // Start New PresentationRequest.onconnectionavailable Test (success) - begin
+    // --------------------------------------------------------------------------
+    var startPresentation = function () {
+        async_test(function (t) {
+            request.onconnectionavailable = t.step_func_done(function (evt) {
+                var connection = evt.connection;
+
+                // Check the initial state of a presentation connection
+                test(function() {
```

Comment:
I haven't checked, but I'm not sure it's good practice to nest calls to `async_test` and `test`, especially because it means the number of test outcomes that appear in the final result depends on the context. For instance, here, if a timeout occurs before the `connectionavailable` event is fired, only the async test line will appear as `FAIL`. The other tests will simply not show up.

I would drop the `test` lines here and rather complete the `assert_xxx` statements with the failure reason as in:
`assert_equals(connection.state, 'connecting', 'The initial state of the presentation connection should be "connecting".');`